src/freertos/posix/src/freertos_hooks.c

```diff
@@ -0,0 +1,32 @@
+#include <FreeRTOS.h>
```

Comment:
Can we use the same hooks fro posix and esp?

Reply:
other thans the idle hook since posix needs a nanosleep

Reply:
I looked into it. I am honestly unsure as to how it will behave to directly register these hooks. Espressif provides an api to set hooks to the api, and recommands using it due to the dual-core nature of the system. 

Maybe they could be shared, but if we ever set them, I would use the api. I could add registration of a simple hook to the build if you deem it necessary, otherwise I would leave this as is.

(https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/freertos_additions.html#hooks)

Reply:
Ahhhh I see, they are supposed to be user hooks, but yea probably espressif is doing something with those hooks since you know, it's espressif (and since it's a dual core). Let's keep the way you've done it. 👍